File: 2984486(small)/nirmesh91/5752104073297920/0/extracted/shuffle.py
T = int(input())
for x in range(0,T):
    N = int(input())
    inp = [int(y) for y in input().split(' ')]
    flag = 0
    for i in range(0,len(inp)):
        if inp[i] < i:
            flag = 1
            break

    if flag == 0:
        print('Case #' + str(x+1) + ': ' + 'BAD')
    else:
        print('Case #' + str(x+1) + ': ' + 'GOOD' )

File: 2984486(small)/MoOT.Cube/5752104073297920/0/extracted/Shuffle.scala
package shuffle

import collection.JavaConversions._
import collection.JavaConverters._
import collection.mutable.ListBuffer

object Shuffle{
  def read(file: String) = {
    val s = io.Source.fromFile(file)
    val c = s.getLines.toList.tail.grouped(2).map{_(1).split(" ").toList.map{_.toInt}}
    s.close
    c
  }

  def write(output: Iterator[String], file: String){
    val buf = new java.io.BufferedWriter(new java.io.FileWriter(file))
    val res = output.zipWithIndex.map{case (v, i) => s"Case #${i+1}: $v"}.mkString("\n")
    println(res)
    buf.write(res)
    buf.close
  }

  def solve(input: ListBuffer[Int], target: List[Int], k: Int): Int = {
    if(k >= input.size) return 0
    val n = if(input(k) == target(k)) 1 else 0
    val j = input.indexOf(target(k))
    val tmp = input(j)
    input.update(j, input(k))
    input.update(k, tmp)
    n + solve(input, target, k + 1)
  }

  def solve(input: List[Int]): String = {
    if(solve(ListBuffer(input.sorted : _*), input, 0) > math.log(input.size)) "BAD" else "GOOD"
  }

  def main(args: Array[String]){
    write(read(args(0)).map{solve}, "outshuffle")
  }

}
